### PR TITLE
jupyter test + automatic logger tagging

### PR DESCRIPTION
**Pull request by mmaecki**
Solves https://github.com/SebChw/art/issues/78 and https://github.com/SebChw/art/issues/73.
However for an unknown to me reason suddenly step 7 in tutorial (overfit) fails, which souldn't be a case. Therefore the test will fail as well.
Exception: Step MNISTModel_7_Overfit failed: Check failed for step: Overfit. Reason: Score 0.8349999785423279 is not greater than 0.88

### CODE REVIEW

File: tests/test_jupyter_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+
+def download_tutorial():
+    try:
+        cookiecutter(
+            "https://github.com/SebChw/art_template.git",
+            no_input=True,
+            extra_context={"project_name": "mnist_tutorial", "author": "test", "email": "test"},  # Pass the project_name to the template,
+            checkout="mnist_tutorial_cookiecutter",  # Use the latest version of the template
```
Comment (kordc) [inline tests/test_jupyter_tutorial.py:18]: IDK if this is the way we want it. If so, we'd need to first merge a new version of the tutorial and then merge this one. Maybe the script should look into the comments look for the art_template link and then take the branch from the link.

File: tests/test_jupyter_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        )
+    except Exception as e:
+        print("Error while generating project using Cookiecutter:", str(e))
+        raise e
+    return 0
```
Comment (kordc) [inline tests/test_jupyter_tutorial.py]: This function shouldn't return anything

File: tests/test_jupyter_tutorial.py
Comment (kordc) [inline tests/test_jupyter_tutorial.py:1]: Create docstrings!

File: tests/utils/augmentations_0.png
Comment (kordc) [inline tests/utils/augmentations_0.png:1]: I think we shouldn't have those images hardcoded in art

File: tests/utils/checkpoints/AlreadyExistingSolutionBaseline_4_Evaluate Baseline/results.json
Comment (kordc) [inline tests/utils/checkpoints/AlreadyExistingSolutionBaseline_4_Evaluate Baseline/results.json:1]: also, those checkpoints shouldn't be here. If you need them to be saved in the art directory, not in the project, maybe add a corresponding .gitignore?

File: tests/utils/checkpoints/state.json
Comment (kordc) [inline tests/utils/checkpoints/state.json:1]: same for it, and every file in checkpoints or utils

**Review by kordc — DISMISSED** [review]

**Comment by kordc** [comment]
Also, as in https://github.com/SebChw/art/issues/73#issuecomment-1766308144 it'd be good to delete those files in this PR as well, as you say it solves #73

File: tests/test_jupyter_tutorial.py (hunk 2 above)
Comment (mmaecki) [reply tests/test_jupyter_tutorial.py]: Right, i was just checking sth and forgot to remove it

File: tests/test_jupyter_tutorial.py (hunk 1 above)
Comment (mmaecki) [reply tests/test_jupyter_tutorial.py:18]: Not sure about it as well. However we should remember that this is a test of ART not the test of a notebook. At least this is what I was aiming to achieve.

File: tests/utils/augmentations_0.png
Comment (trebacz626) [reply tests/utils/augmentations_0.png:1]: Yep

**Comment by trebacz626** [comment]
Good, but I wouldn't use store actual files in tests, maybe except jsons. Just if there is a png check if it exists and whether it really is a png file.

**Comment by trebacz626** [comment]
I think we should have a list of files to check for existance. ideally we would also check if png file is an image but lets not waste time on it.

**Review by SebChw — APPROVED** [review]

**Pull request merged**